examples/car_control/run.py

```diff
@@ -0,0 +1,101 @@
+    gameplay_statics_default_object = instance.unreal_service.get_default_object(uclass=gameplay_statics_class, create_if_needed=False)
+
+    set_game_paused_func = instance.unreal_service.find_function_by_name(uclass=gameplay_statics_class, name="SetGamePaused")
+
+    get_player_controller_func = instance.unreal_service.find_function_by_name(uclass=gameplay_statics_class, name="GetPlayerController")
+    get_player_controller_return_values = instance.unreal_service.call_function(uobject=gameplay_statics_default_object, ufunction=get_player_controller_func,
+                                                                                args={"PlayerIndex": 0})
+    player_controller = instance.unreal_service.to_handle(get_player_controller_return_values["ReturnValue"])
+    controller_class = instance.unreal_service.load_class(class_name="UObject", outer=0, name="/Script/Engine.Controller", filename="")
+
+    enhanced_input_component_class = instance.unreal_service.load_class(class_name="UObject", outer=0, name="/Script/EnhancedInput.EnhancedInputComponent", filename="")
+    enhanced_input_component = instance.unreal_service.get_component_by_class(player_controller, enhanced_input_component_class)
+
+    vehicle_movement_component_class = instance.unreal_service.load_class(class_name="UObject", outer=0, name="/Script/ChaosVehicles.ChaosVehicleMovementComponent", filename="")
+
+    # spawn a blueprint actor
+    agent_uclass = instance.unreal_service.load_class(class_name="UObject", outer=0, name=args.bp_class_reference, filename="")
+    agent = instance.unreal_service.spawn_actor_from_uclass(
+        uclass=agent_uclass,
+        location={"X": 0.0, "Y": 300.0, "Z": 130.0}, rotation={"Roll": 0.0, "Pitch": 0.0, "Yaw": 0.0},
+        spawn_parameters={"Name": "Agent", "SpawnCollisionHandlingOverride": "AlwaysSpawn"}
+    )
+
+    instance.engine_service.tick()
+    instance.engine_service.end_tick()
+
+    instance.engine_service.begin_tick()
+
+    movement_component = instance.unreal_service.get_component_by_class(agent, vehicle_movement_component_class)
+    set_throttle_input_func = instance.unreal_service.find_function_by_name(uclass=vehicle_movement_component_class, name="SetThrottleInput")
+    set_steering_input_func = instance.unreal_service.find_function_by_name(uclass=vehicle_movement_component_class, name="SetSteeringInput")
+    instance.unreal_service.set_object_properties_for_uobject(movement_component, {"bRequiresControllerForInputs": False})
+
+    instance.engine_service.tick()
+    instance.engine_service.end_tick()
+
+    spear.log("Executing sequence of actions as provided in the actions file...")
+
+    frame = 0
+    start_time_seconds = time.time()
+    while frame < 200:
+        instance.engine_service.begin_tick()
+        instance.unreal_service.call_function(uobject=gameplay_statics_default_object, ufunction=set_game_paused_func, args={"bPaused": False})
+
+        if frame < 300 and frame >= 10:
```

Comment:
This logic is non-sensical because `frame` is guaranteed to always be less than 300 inside this loop. Why are we using a while loop instead of a for loop? I think it would be cleaner to just have a few for loops that execute sequentially.